core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlRelationBuilder.scala

```diff
@@ -10,9 +10,44 @@ class TSqlRelationBuilder extends TSqlParserBaseVisitor[ir.Relation] with Incomp
   protected override def wrapUnresolvedInput(unparsedInput: String): ir.Relation = ir.UnresolvedRelation(unparsedInput)
 
   override def visitTableSourceItem(ctx: TableSourceItemContext): ir.Relation = {
-    val table = ctx.fullTableName().getText
     // [TODO]: Handle Table Alias ctx.tableAlias().getText
-    ir.NamedTable(table, Map.empty, is_streaming = false)
+    ctx.fullTableName().accept(this)
+  }
+
+  override def visitFullTableName(ctx: FullTableNameContext): ir.NamedTable = {
+    // Extract the components of the full table name, if they exist
+    val linkedServer = Option(ctx.linkedServer).map(_.getText)
+    val database = Option(ctx.database).map(_.getText)
+    val schema = Option(ctx.schema).map(_.getText)
+    val name = ctx.table.getText
+
+    // Build the unparsed_identifier string
+    val unparsedIdentifier = List(linkedServer, database, schema, Some(name)).flatten.mkString(".")
+
+    // Create the NamedTable
+    ir.NamedTable(unparsedIdentifier, Map.empty, is_streaming = false)
+  }
+
+  /**
+   * Note that SELECT a, b, c FROM x, y, z is equivalent to SELECT a, b, c FROM x CROSS JOIN y CROSS JOIN z
+   * @param ctx
+   *   the parse tree
+   */
+  override def visitTableSources(ctx: TSqlParser.TableSourcesContext): ir.Relation = {
+    val relations = ctx.tableSource().asScala.toList.map(_.accept(this)).collect { case r: ir.Relation => r }
+    relations match {
+      case head :: Nil => head
```

Comment:
```suggestion
```
This case isn't needed as it is covered by the `foldLeft` in the case bellow

Reply:
I will check that and remove it if it can indeed be removed.